src/editor/mjml/index.js

```diff
@@ -105,19 +111,28 @@ function MJML() {
 				.finally( () => {
 					unlockPostSaving( 'newspack-newsletters-refresh-html' );
 					setIsRefreshingHTML( false );
-					// Check for sync errors after refreshing the HTML.
-					apiFetch( {
-						path: `/newspack-newsletters/v1/${ postId }/sync-error`,
-					}).then( ( { error_message } ) => {
-						if ( error_message ) {
-							createNotice( 'error', error_message, {
-								id: 'newspack-newsletters-newsletter-sync-error',
-								isDismissible: true,
-							} );
-						} else {
-							removeNotice( 'newspack-newsletters-newsletter-sync-error' );
-						}
-					} );
+
+					const { name: serviceProviderName } = getServiceProvider();
+					const { supported_esps: suppportedESPs } = newspack_email_editor_data || {};
+					const isSupportedESP = serviceProviderName && 'manual' !== serviceProviderName && suppportedESPs?.includes( serviceProviderName );
```

Comment:
Whoops, it's actually an array, not an object, I was just setting the wrong fallback value. be52969 should fix this.

Reply:
In my case the newspack_email_editor_data.supported_esps is also coming as an object:

```
{
  '0': 'mailchimp',
  '1': 'constant_contact',
  '3': 'active_campaign',
  '4': 'manual'
}
```

That's because of this:
https://github.com/Automattic/newspack-newsletters/blob/a22b7219cbaaaea6e69175bfa42307e4a867cb7e/includes/class-newspack-newsletters.php#L123-L126

`array_diff()` won't reset the indexes and json encoding an indexed array with a skipped index will transform it into an object.

An `array_values()` there should suffice.

Also, there's an extra P in `suppportedESPs`.

Reply:
Ah, I see what you mean! I wasn't getting the object because I had the `NEWSPACK_NEWSLETTERS_SUPPORT_DEPRECATED_CAMPAIGN_MONITOR` constant enabled, so the array wasn't getting passed through `array_diff`. c00aca5 should fix both issues. :ppp